components/clp-package-utils/clp_package_utils/scripts/native/decompress.py

```diff
@@ -127,7 +136,12 @@ def handle_extract_stream_cmd(
             orig_file_id = parsed_args.orig_file_id
         else:
             orig_file_path = parsed_args.orig_file_path
-            orig_file_id = get_orig_file_id(clp_config.database, orig_file_path)
+            orig_file_id = get_orig_file_id(
+                clp_config.database,
+                clp_config.package.storage_engine,
+                CLP_DEFAULT_DATASET_NAME,
+                orig_file_path,
+            )
```

Comment:
Technically, this logic is only supported for the clp storage engine. But I guess it doesn't hurt to support datasets for it if we support it for the clp-s storage engine in the future.

That said, we need to add the `dataset` field to `ExtractJsonJobConfig` and handle using it wherever `ExtractJsonJobConfig`. Sorry, I should've caught this in the previous PR. Currently, clicking on the "Original File" link in the UI search results will open a page which never finishes loading. Once we implement JSON extraction correctly, that should work.